client/state/ui/first-view/selectors.js

```diff
@@ -0,0 +1,51 @@
+/** @ssr-ready **/
+
+/**
+ * External dependencies
+ */
+import filter from 'lodash/filter';
+import some from 'lodash/some';
+import startsWith from 'lodash/startsWith';
+import takeRightWhile from 'lodash/takeRightWhile';
+
+/**
+ * Internal dependencies
+ */
+import { FIRST_VIEW_START_DATES } from './constants';
+import { getActionLog } from 'state/ui/action-log/selectors';
+import { getPreference } from 'state/preferences/selectors';
+import { getSectionName, isSectionLoading } from 'state/ui/selectors';
+import { isEnabled } from 'config';
+
+export function doesViewHaveFirstView( view ) {
+	return !! ( FIRST_VIEW_START_DATES[ view ] );
+}
+
+export function isViewEnabled( state, view ) {
+	const firstViewHistory = getPreference( state, 'firstViewHistory' ).filter( entry => entry.view === view );
+	const latestFirstViewHistory = [ ...firstViewHistory ].pop();
+	const isViewDisabled = latestFirstViewHistory ? ( !! latestFirstViewHistory.disabled ) : false;
+	return doesViewHaveFirstView( view ) && ! isViewDisabled;
+}
+
+export function wasViewHidden( state, view ) {
+	return -1 !== state.ui.firstView.hidden.indexOf( view );
+}
+
+export function switchedFromDifferentSection( state ) {
+	const section = state.ui.section;
+	const routeSets = filter( getActionLog( state ), entry => entry.type === 'ROUTE_SET' );
```

Comment:
We should be using the proper `ROUTE_SET` constant from `action-types`, rather than the string representation.

While we're here: if lodash's `filter` is only used in its `filter( collection, predicate )` form, then it could be replaced with the native implementation: `getActionLog( state ).filter( entry => … )`. An other usage would be the convenient: `filter( getActionLog( state ), { type: ROUTE_SET } )`.